src/sentry/models/userip.py

```diff
@@ -26,21 +32,53 @@ class Meta:
     __repr__ = sane_repr("user_id", "ip_address")
 
     @classmethod
-    def log(cls, user, ip_address):
+    def log(cls, user: User, ip_address: str):
         # Only log once every 5 minutes for the same user/ip_address pair
         # since this is hit pretty frequently by all API calls in the UI, etc.
         cache_key = f"userip.log:{user.id}:{ip_address}"
-        if cache.get(cache_key):
-            return
+        if not cache.get(cache_key):
+            _perform_log(user, ip_address)
+            cache.set(cache_key, 1, 300)
+
 
-        try:
-            geo = geo_by_addr(ip_address)
-        except Exception:
-            geo = None
+def _perform_log(user: User, ip_address: str):
+    try:
+        geo = geo_by_addr(ip_address)
+    except Exception:
+        geo = None
 
-        values = {"last_seen": timezone.now()}
-        if geo:
-            values.update({"country_code": geo["country_code"], "region_code": geo["region"]})
+    values = {"last_seen": timezone.now()}
+    if geo:
+        values.update({"country_code": geo["country_code"], "region_code": geo["region"]})
 
+    user.get_orgs()
+    if SiloMode.get_current_mode() == SiloMode.REGION:
+        _perform_kafka_log(user, ip_address, values)
+    else:
         UserIP.objects.create_or_update(user=user, ip_address=ip_address, values=values)
```

Comment:
@RyanSkonnord was considering an abstraction that would handle this switching for us, which might be a good idea. 

One abstraction I've used before is a function that is both the dispatcher and the handler of an event, so you can call it directly in a context like this `create_or_update_user_ip()` to enqueue the kafka message and the kafka consumer implementation looks identical `create_or_update_user_ip()` but does the actual work because it's in the right silo. It makes for quite readable code, but does hide some complexity 🤷

Reply:
I don't have a strong opinion, but I think it will be easier to form a better opinion _after_ I add more of these region -> control silo code paths, because I think there will be more complexity in the audit log code paths.

Also, to be clear, is there some general concern having the control/region silos behave differently generally?  I expect both code paths to be tested and neither to be particularly complex.